### PR TITLE
Parameterize ITs to run with concurrent search enabled

### PR DESCRIPTION
**Pull request by harishbhakuni**
### Description
Parameterizing first batch of ITs with concurrent search enabled as part of this PR.
List of ITs Parameterized in this PR:
```
1. IndicesRequestCacheIT
2. IndexStatsIT
3. ScriptCacheIT
4. StressSearchServiceReaperIT
5. DeletePitMultiNodeIT
6. PitMultiNodeIT
7. ScriptQuerySearchIT
8. DuelScrollIT
9. SearchScrollIT
10. SearchScrollWithFailingNodesIT
11. SearchAfterIT
12. GeoDistanceIT
13. GeoDistanceSortBuilderIT
14. SortFromPluginIT
15. CompletionSuggestSearchIT
16. ContextCompletionSuggestSearchIT
17. SuggestSearchIT
18. SimilarityIT
```
### Related Issues
#9443 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/internalClusterTest/java/org/opensearch/search/sort/FieldSortIT.java [hunk 1]
```diff
@@ -105,7 +109,24 @@
 import static org.hamcrest.Matchers.nullValue;
 import static org.hamcrest.Matchers.oneOf;
 
-public class FieldSortIT extends OpenSearchIntegTestCase {
+public class FieldSortIT extends ParameterizedOpenSearchIntegTestCase {
```
Comment (neetikasinghal) [inline server/src/internalClusterTest/java/org/opensearch/search/sort/FieldSortIT.java:112]: I think we can skip this one as this has some tests failures and I am fixing it as part of https://github.com/opensearch-project/OpenSearch/issues/8510. I am planning to enable the parameterization as part of the pr that contains the fix.

File: server/src/internalClusterTest/java/org/opensearch/threadpool/SimpleThreadPoolIT.java [hunk 2]
```diff
@@ -32,26 +32,50 @@
...
 import java.util.Set;
 import java.util.regex.Pattern;
 
 import static org.opensearch.common.xcontent.XContentFactory.jsonBuilder;
+import static org.opensearch.search.SearchService.CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING;
 import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertNoFailures;
 
 @ClusterScope(scope = Scope.TEST, numDataNodes = 0, numClientNodes = 0)
-public class SimpleThreadPoolIT extends OpenSearchIntegTestCase {
+public class SimpleThreadPoolIT extends ParameterizedOpenSearchIntegTestCase {
```
Comment (neetikasinghal) [inline server/src/internalClusterTest/java/org/opensearch/threadpool/SimpleThreadPoolIT.java:61]: I don't think we need parameterization for this class. Also, the test in this class looks to be failing as part of the gradle check

File: server/src/internalClusterTest/java/org/opensearch/cluster/settings/ClusterSettingsIT.java [hunk 3]
```diff
@@ -43,23 +45,43 @@
...
+import static org.opensearch.search.SearchService.CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING;
 import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertAcked;
 import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertBlocked;
 import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.notNullValue;
 import static org.hamcrest.Matchers.nullValue;
 
-public class ClusterSettingsIT extends OpenSearchIntegTestCase {
+public class ClusterSettingsIT extends ParameterizedOpenSearchIntegTestCase {
```
Comment (neetikasinghal) [inline server/src/internalClusterTest/java/org/opensearch/cluster/settings/ClusterSettingsIT.java:67]: i dont think we need parameterization here as it has tests specific to checking the other settings, looks unrelated to the concurrent search use-case.

File: server/src/internalClusterTest/java/org/opensearch/search/scroll/SearchScrollWithFailingNodesIT.java [hunk 4]
```diff
@@ -32,26 +32,50 @@
...
+import static org.opensearch.search.SearchService.CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING;
 import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertAcked;
 import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertAllSuccessful;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.greaterThan;
 import static org.hamcrest.Matchers.lessThan;
 
 @OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST, numDataNodes = 0, numClientNodes = 0)
-public class SearchScrollWithFailingNodesIT extends OpenSearchIntegTestCase {
+public class SearchScrollWithFailingNodesIT extends ParameterizedOpenSearchIntegTestCase {
```
Comment (neetikasinghal) [inline server/src/internalClusterTest/java/org/opensearch/search/scroll/SearchScrollWithFailingNodesIT.java:61]: there are test failures in this class well, please check.

**Comment by neetikasinghal** [comment]
@harishbhakuni thanks for the changes, it would be good to add the classes for which we are adding the parameterization in the pr description.

File: server/src/internalClusterTest/java/org/opensearch/script/StoredScriptsIT.java [hunk 5]
```diff
@@ -31,21 +31,41 @@
...
 import java.util.Collection;
 import java.util.Collections;
 import java.util.Map;
 import java.util.function.Function;
 
+import static org.opensearch.search.SearchService.CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING;
 import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertAcked;
 
-public class StoredScriptsIT extends OpenSearchIntegTestCase {
+public class StoredScriptsIT extends ParameterizedOpenSearchIntegTestCase {
```
Comment (sohami) [inline server/src/internalClusterTest/java/org/opensearch/script/StoredScriptsIT.java:52]: Not needed for this class as well

File: server/src/internalClusterTest/java/org/opensearch/search/sort/FieldSortIT.java (hunk 1 above)
Comment (harishbhakuni) [reply server/src/internalClusterTest/java/org/opensearch/search/sort/FieldSortIT.java]: makes sense, ack.

File: server/src/internalClusterTest/java/org/opensearch/indices/stats/IndexStatsIT.java [hunk 6]
```diff
@@ -116,7 +120,23 @@
...
+        return Arrays.asList(
+            new Object[] { Settings.builder().put(CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING.getKey(), false).build() },
+            new Object[] { Settings.builder().put(CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING.getKey(), true).build() }
+        );
+    }
+
+    @Override
+    protected Settings featureFlagSettings() {
+        return Settings.builder().put(super.featureFlagSettings()).put(FeatureFlags.CONCURRENT_SEGMENT_SEARCH, "true").build();
+    }
```
Comment (jed326) [inline server/src/internalClusterTest/java/org/opensearch/indices/stats/IndexStatsIT.java:139]: @harishbhakuni do you mind if I do this as a part of #9622? I want to add some tests specific to concurrent search here.

Comment (jed326) [reply server/src/internalClusterTest/java/org/opensearch/indices/stats/IndexStatsIT.java:139]: I didn't end up picking this up as a part of the PR above so it still needs parameterization

Comment (harishbhakuni) [reply server/src/internalClusterTest/java/org/opensearch/indices/stats/IndexStatsIT.java:139]: @jed326 sorry for delayed response. just saw this thread.
> I didn't end up picking this up as a part of the PR above so it still needs parameterization

oh okay, sure i will update it as part of this PR.

File: server/src/internalClusterTest/java/org/opensearch/search/pit/DeletePitMultiNodeIT.java [hunk 7]
```diff
@@ -52,7 +59,23 @@
  * Multi node integration tests for delete PIT use cases
  */
 @OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.SUITE, numDataNodes = 2)
-public class DeletePitMultiNodeIT extends OpenSearchIntegTestCase {
+public class DeletePitMultiNodeIT extends ParameterizedOpenSearchIntegTestCase {
```
Comment (sohami) [inline server/src/internalClusterTest/java/org/opensearch/search/pit/DeletePitMultiNodeIT.java:62]: This class doesn't need parameterization as it is verifying deletion of Pits only

**Comment by sohami** [comment]
@harishbhakuni There is a conflict with `server/src/internalClusterTest/java/org/opensearch/search/sort/GeoDistanceIT.java`, can you please resolve that

**Comment by harishbhakuni** [comment]
below tests are failing in gradle check and are marked flaky (not able to reproduce them in local):
org.opensearch.search.SearchWeightedRoutingIT.testSearchAggregationWithNetworkDisruption_FailOpenEnabled: #5957 
org.opensearch.indices.replication.SegmentReplicationSuiteIT.classMethod: #9499 
org.opensearch.index.ShardIndexingPressureSettingsIT.testShardIndexingPressureEnforcedEnabledDisabledSetting: #2348

**Review by sohami — APPROVED** [review]
